File: Python/no. checking/karpekar.py
#Karpekar no.
n=int(input("Enter the no. = "))
sq=n**2
s=0
c=len(str(n))
s=sq%10**c+sq//10**c
if n==s:
    print("Karpekar no.")
else:
    print("Not Karpekar No.")

File: C Programms/No. Cheking In Range/karpekar.c
//WAP to print Karpekar no. for 1-n
#include<stdio.h>
void main()
{
    int n,r,s,x,p,sq;
    printf("Enter the range = ");
    scanf("%d",&r);
    printf("Karpekar No. from 1 to %d are ...\n",r);
    for(n=1;n<=r;n++)
    {
        for(x=n,sq=n*n,p=1;x!=0;x/=10)
            p*=10;
        if(sq%p+sq/p == n)
            printf("%d ",n);
    }
}